opentelemetry/src/api/trace/mod.rs

```diff
@@ -138,12 +139,50 @@ pub use self::{
     },
     tracer::{SpanBuilder, Tracer},
 };
+use crate::exporter::trace::ExportError;
+use std::time;
+
+/// Describe the result of operations in tracing API.
+pub type TraceResult<T> = Result<T, TraceError>;
 
 /// Errors returned by the trace API.
-#[derive(Error, Debug, PartialEq)]
+#[derive(Error, Debug)]
 #[non_exhaustive]
 pub enum TraceError {
     /// Other errors not covered by specific cases.
     #[error("Trace error: {0}")]
     Other(String),
+
+    /// Export failed with the error returned by the exporter
+    #[error("Exporting failed with {0}")]
+    ExportFailed(Box<dyn ExportError>),
+
+    /// Export failed to finish after certain period and processor stopped the export.
+    #[error("Exporting timed out after {} seconds", .0.as_secs())]
+    ExportTimedOut(time::Duration),
+
+    /// Other errors propagated from trace SDK that weren't covered above
+    #[error(transparent)]
+    Boxed(#[from] Box<dyn std::error::Error + Send + Sync + 'static>),
```

Comment:
Could we combine this with `Other`? could look at `std::io::Error` or other crates to see best way of having catchall error type